ctoken-monitor/src/autotask.spec.js

```diff
@@ -0,0 +1,175 @@
+            scanNodeCount
+            severity
+            source {
+                transactionHash
+                agent {
+                    id
+                }
+                block {
+                    chainId
+                    hash
+                }
+            }
+            findingType
+          }
+        }
+      }`,
+    variables: {
+      input: {
+        first: 100,
+        agents: [agentId],
+        blockNumberRange: {
+          startBlockNumber,
+          endBlockNumber,
+        },
+        createdSince: 0,
+        chainId: 1,
+      },
+    },
+  };
+
+  // perform the POST request
+  const response = await axios({
+    url: fortaApiEndpoint,
+    method: 'post',
+    headers,
+    data: graphqlQuery,
+  });
+
+  const { data } = response;
+  if (data === undefined) {
+    return undefined;
+  }
+
+  const { data: { alerts: { alerts } } } = data;
+  return alerts;
+}
+
+async function createFortaSentinelEvents(agentId, startBlockNumber, endBlockNumber) {
+  const alerts = await getFortaAlerts(agentId, startBlockNumber, endBlockNumber);
+  const autotaskEvents = alerts.map((alert) => {
+    // augment the alert Object with additional fields
+    // admittedly, there is some hand-waving here because we have to mock some of the Sentinel
+    // fields that don't originate from the Forta Public API
+    // e.g. We have to specify the alertId in the Sentinel to perform filtering on what we get from
+    // the Forta Agent in the first place.
+    /* eslint-disable no-param-reassign */
+    alert.source.agent.name = 'N/A';
+    alert.source.block.chain_id = alert.source.block.chainId;
+    alert.source.tx_hash = alert.source.transactionHash;
+    alert.alertType = 'TX';
+    alert.alert_id = 'ALERT_ID_PLACEHOLDER';
+    alert.type = 'INFORMATION';
+    alert.scanner_count = 1;
+    /* eslint-enable no-param-reassign */
+
+    // populate the matchReasons Array with placeholders
+    const matchReasons = [
+      {
+        type: 'alert-id',
+        value: 'ALERT_ID_PLACEHOLDER',
+      },
+      {
+        type: 'severity',
+        value: 'INFO',
+      },
+    ];
+
+    // populate the sentinel Object with placeholders
+    // none of these are currently checked by any Autotasks in use
+    const sentinel = {
+      id: '8fe3d50b-9b52-44ff-b3fd-a304c66e1e56',
+      name: 'Sentinel Name Placeholder',
+      addresses: [],
+      agents: [agentId],
+      network: 'mainnet',
+      chainId: 1,
+    };
+
+    const autotaskEvent = {
+      relayerARN: undefined,
+      kvstoreARN: undefined,
+      credentials: undefined,
+      tenantId: undefined,
+      secrets,
+      request: {
+        body: {
+          hash: alert.hash, // forta Agent hash
+          alert,
+          matchReasons,
+          sentinel,
+          type: 'FORTA',
+        },
+      },
+    };
+    return autotaskEvent;
+  });
+
+  return autotaskEvents;
+}
+
+it('Runs autotask against blocks in configuration file', async () => {
```

Comment:
Task runs and posts on discord but the test reports `Failure`.  This is the error message `Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.`

Reply:
I've tested with extending the timeout and it still fails. Actually running the normal `npm test` returns successful at first until it too times out. I'm not sure if fixing this issue is within your scope.

Reply:
I looked into this a bit more as well and didn't find a proper solution. I can make a note or a small ticket about it, but it looks ok to me if it's firing alerts to discord

Reply:
looked into it some more and it works with a smaller range of blocks:

   "startBlockNumber": 14788000,
    "endBlockNumber": 14790000

Reply:
I think I set it to 1 block range and it was still having issues. I'll check it again in a bit. If the smaller block range works, then cool.

Reply:
Yep that works!